src/main/java/com/impactupgrade/nucleus/service/segment/MailchimpCommunicationService.java

```diff
@@ -432,4 +442,25 @@ protected MemberInfo toMcMemberInfo(CrmContact crmContact, Map<String, Object> c
 
     return mcContact;
   }
+
+  private boolean smsAllowed(CrmContact crmContact, EnvironmentConfig.CommunicationPlatform mailchimpConfig) {
+    boolean smsOptIn = Boolean.TRUE == crmContact.smsOptIn && Boolean.FALSE == crmContact.smsOptOut;
+    if (!smsOptIn) {
+      return false;
+    }
+    String phoneNumber = crmContact.phoneNumberForSMS();
+    if (Strings.isNullOrEmpty(phoneNumber)) {
+      return false;
+    }
+
+    boolean smsAllowed = false;
+    if (!Strings.isNullOrEmpty(mailchimpConfig.countryCode) && phoneNumber.startsWith(mailchimpConfig.countryCode)) {
+      smsAllowed = true;
+    } else if (!phoneNumber.startsWith("+") && !Strings.isNullOrEmpty(mailchimpConfig.country)
+        && crmContact.account.billingAddress != null && !Strings.isNullOrEmpty(crmContact.account.billingAddress.country)
+        && crmContact.account.billingAddress.country.equalsIgnoreCase(mailchimpConfig.country)) {
```

Comment:
Solid! One suggestion: let's also check `crmContact.account.mailingAddress` and `crmContact.mailingAddress`? Just in case. BillingAddress is more typical, but we do periodically have situations where the others are used. If Australia is in either of the 3, smsAllowed.